### PR TITLE
feat: add Nightwatch.js as test framework option

### PR DESCRIPTION
**Pull request by astagi**
This PR adds [Nightwatch.js](https://nightwatchjs.org/) as test framework option 🦉

### CODE REVIEW

**Comment by Atinux** [comment]
Do you mind resolving the conflicts so we can merge it?

**Comment by astagi** [comment]
No problem @Atinux I'll check it this evening or tomorrow and rebase this branch with master resolving the conflict.

**Comment by astagi** [comment]
@Atinux @clarkdo conflicts resolved 👍 (tests on macos lastest 12 timed out, try to restart the CI process)

**Comment by astagi** [comment]
@Atinux @clarkdo @pi0 I've just rebased this branch with master and updated Nightwatch.js, unfortunately I get some errors on snapshots 😞

File: renovate.json [hunk 1]
```diff
@@ -57,6 +57,10 @@
     {
       "packagePatterns": ["wdio", "webdriver"],
       "groupName": "webdriverio packages"
+    },
+    {
+      "packagePatterns": ["nightwatch"],
```
Comment (clarkdo) [inline renovate.json:62]: Should `chromedriver` be included ?

**Review by clarkdo — APPROVED** [review]

File: renovate.json (hunk 1 above)
Comment (astagi) [reply renovate.json:62]: yes, it needs at least one driver to run tests, `chromedriver` is the most common one.

**Comment by clarkdo** [comment]
@astagi I've updated snapshot job, can you please merge maseter and try ?

**Comment by astagi** [comment]
> @astagi I've updated snapshot job, can you please merge latest master and try ?

Thank you @clarkdo 👍 I've just rebased my branch!

File: renovate.json (hunk 1 above)
Comment (astagi) [reply renovate.json]: @clarkdo sorry I replied too fast here! I've just added chromedriver in renovate.json, I forgot to add it before!

**Comment by clarkdo** [comment]
Thanks! @astagi

**Pull request merged**